### ISSUE TITLE
Suggested citations of FR lessons

### ISSUE THREAD

**Issue by spapastamkou** [programminghistorian/jekyll#2553]
The suggested citations of FR lessons - at the end of the page of each lesson - go as follows: 

Jon Crump, «Générer un jeu de données structuré à partir d’un texte océrisé,» traduction par Antoine Gourlay, *Programming Historian en français* 3 (2021), https://doi.org/10.46430/phfr0020

There are two small problems here:
1) the comma after the title is inside the FR quotation marks - just like in the EN lessons but not the ES and PT ones (I think it comes from the Chicago style citation of bibliographical references)
2) there is no non-breaking space after the left opening quotation mark and before the right closing quotation mark

In short, this
«Générer un jeu de données structuré à partir d’un texte océrisé,»
should rather look like this ideally
«&#x202F;Générer un jeu de données structuré à partir d’un texte océrisé&#x202F;»,

Is there a way to remedy to this @programminghistorian/technical-team ? I have explored a bit in the past the folders on `jekyll` to see where all this comes from but did not find. Thank you! 

Thanks @matgille for having reminded this.

**Comment by spapastamkou** [comment]
cc @anisa-hawes

**Comment by jenniferisasi** [comment]
Thanks for pointing this out @spapastamkou. The technical team actually learned a few weeks back about the spaces after/before certain punctuation in French thanks to @anisa-hawes. This looks like a little interesting fix so I am going to try to fix it.

**Comment by spapastamkou** [comment]
Thanks @jenniferisasi. I'll forward you all relevant information that I sent also to Anisa about encoding the special FR characters, the quotation marks, left and right, included (I did not use these in the markdown of the initial comment. I would be interested to know where you fix this, once you are done.

**Comment by jenniferisasi** [comment]
Sure thing! The process I followed to find where to make the change (cos I had a slight idea...) was to check first the `_layouts/lesson.html` file and look for the code for citations, in particular the part that creates the text of the citation: 

```
<div class="suggested-citation-text">
      <p class="suggested-citation-text">{% include author.html %},
        {{ site.data.snippets.title-open-quote[page.lang] }}{{ page.title | strip }}{{ site.data.snippets.title-close-quote[page.lang] }}{% if page.translator %}
        {{ site.data.snippets.translator[page.lang] }}
        {{ page.translator | array_to_sentence_string: site.data.snippets.and[page.lang] }},{% endif %}
        <em>{{ site.data.snippets.journal-title[page.lang] }}</em> {{ vol_no }} ({{ page_year }}),
        https://doi.org/{{ page.doi }}.</p>
    </div>
```

This confirmed that the changed had to be done in `_data/snippets.yml` for the `title-open-quote` and `title-close-quote` that varies per the `[page.lang]` that is the language in which the page is in. So I went to snippets and changed: 
```  
fr: '«'
fr: ',»'
```
to 
``` 
fr: '« '
fr: ' »,'
```

**Comment by spapastamkou** [comment]
Well, I drop this here (please check the raw markdown by editing this comment): 

« => &laquo; 
» => &raquo;

(I use the html entities but alternatively the ISO codes can work as well, you can find them here if necessary: https://www.lehtml.com/html/speciaux.htm)

For the non-breaking space, used before several punctuation signs (: ; ! ? ») and after «, there are at least 2 solutions: 
&nbsp; (in html) => this one is full
&#x202F; (Unicode) => this is a bit narrower (something like half-space)

I preferably use the second one which corresponds better to the FR use (as we find it in Word etc) but the other is ok as well.  

If necessary, you can find examples of extensive use of these codes in the following FR lesson files on jekyll: 
calibration-radiocarbone-avec-r.md
detecter-la-reutilisation-de-texte-avec-passim.md

**Comment by jenniferisasi** [comment]
Oh, I see! It is not a regular space but a narrower one 🤔 let me see how I add that in the snippets in #2554

**Comment by spapastamkou** [comment]
Ok, thanks a lot! I remember I had put my nose only in ` _data/snippets.yml` and of course I did not spot these elements as I had not checked the other file before. 

We can keep the regular space as it is if too difficult to add the code, the most important was the comma:)

**Comment by jenniferisasi** [comment]
I'm going to try to add the right one :)

**Comment by spapastamkou** [comment]
Ok, this was quick:-) Done in ##2554 thanks to @jenniferisasi !

**Comment by jenniferisasi** [comment]
Done!
